azurerm/internal/services/redisenterprise/redis_enterprise_cluster_resource.go

```diff
@@ -0,0 +1,296 @@
+						"1",
+						"2",
+						"3",
+					}, true),
+				},
+			},
+
+			// RP currently does not return this value, but will in the near future (RP defaults to 1.2)
+			// "minimum_tls_version": {
+			// 	Type:     schema.TypeString,
+			// 	Optional: true,
+			// 	Default:  string(redisenterprise.OneFullStopTwo),
+			// 	ValidateFunc: validation.StringInSlice([]string{
+			// 		string(redisenterprise.OneFullStopZero),
+			// 		string(redisenterprise.OneFullStopOne),
+			// 		string(redisenterprise.OneFullStopTwo),
+			// 	}, false),
+			// },
+
+			// RP currently does not return this value, but will in the near future
+			"hostname": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			// RP currently does not return this value, but will in the near future
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tags.ForceNewSchema(),
+		},
+	}
+}
+
+func resourceRedisEnterpriseClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).RedisEnterprise.Client
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+	log.Printf("[INFO] preparing arguments for Redis Enterprise Cluster creation.")
+
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	resourceId := parse.NewRedisEnterpriseClusterID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceId.ResourceGroup, resourceId.Name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for presence of existing Redis Enterprise Cluster (Name %q / Resource Group %q): %+v", resourceId.Name, resourceId.ResourceGroup, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_redis_enterprise_cluster", resourceId.ID())
+		}
+	}
+
+	location := location.Normalize(d.Get("location").(string))
+	sku := expandRedisEnterpriseClusterSku(d.Get("sku_name").(string))
+
+	// If the sku type is flash check to make sure that the sku is supported in that region
+	if strings.Contains(string(sku.Name), "Flash") {
+		if err := validate.RedisEnterpriseClusterFlashSkuTypeLocation(location); err != nil {
+			return fmt.Errorf("%s", err)
+		}
+	}
+
+	parameters := redisenterprise.Cluster{
+		Name:     utils.String(d.Get("name").(string)),
+		Location: utils.String(location),
+		Sku:      sku,
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	if v, ok := d.GetOk("zones"); ok {
+		// Zones are currently not supported in these regions
+		if location == "centraluseuap" || location == "westus" {
+			return fmt.Errorf("Redis Enterprise Cluster (Name %q / Resource Group %q): 'Zones' are not currently supported in the 'West US' or 'Central US EUAP' regions, got %q", resourceId.Name, resourceId.ResourceGroup, location)
```

Comment:
with the generated parse functions i think this can be
```suggestion
			return fmt.Errorf("Redis Enterprise Cluster %q: 'Zones' are not currently supported in the 'West US' or 'Central US EUAP' regions, got %q", resourceId, location)
```

Reply:
Fixed.

Reply:
Why would the Regions be hard coded? Why not reference a lookup or pivot table instead? 
I ask this as the code demonstrated suggests hard coding. The nature of my question is base upon hard coding value adds technical debt to maintain in comparison to a lookup or pivot table containing acceptable configurations.

Reply:
Is this validation occurring at the time of deployment or only after a deployment attempt fails?